applications/sam/sam_rm.py

```diff
@@ -113,26 +114,32 @@ def _removeItems(self, items, args, detectionMethod, filters):
 
         for item in sorted(items):
             itemType = item.getType()
-            toRemove = True
-
-            # sam-rm -d
-            if args.directories and itemType != sequenceParser.eTypeFolder:
-                toRemove = False
-
-            # sam-rm -f
-            if args.files and itemType != sequenceParser.eTypeFile:
-                toRemove = False
+            toRemove = False
 
-            # sam-rm -s
-            if args.sequences and itemType != sequenceParser.eTypeSequence:
-                toRemove = False
+            # sam-rm default case: remove all items
+            if not args.directories and not args.files and not args.sequences:
+                toRemove = True
+            else:
+                # sam-rm -d
+                if args.directories and itemType == sequenceParser.eTypeFolder:
+                    toRemove = True
+                # sam-rm -f
+                elif args.files and itemType == sequenceParser.eTypeFile:
+                    toRemove = True
+                # sam-rm -s
+                elif args.sequences and itemType == sequenceParser.eTypeSequence:
+                    toRemove = True
 
             # sam-rm -R
             if args.recursive and itemType == sequenceParser.eTypeFolder:
-                subFolder = os.path.join(item.getFolder(), item.getFilename())
-                self.logger.debug('Browse in sub folder"' + subFolder + '" with the following filters: ' + str(filters))
-                newItems = sequenceParser.browse(subFolder, detectionMethod, filters)
-                self._removeItems(newItems, args, detectionMethod, filters)
+                try:
+                    subFolder = os.path.join(item.getFolder(), item.getFilename())
+                    self.logger.debug('Browse in sub folder"' + subFolder + '" with the following filters: ' + str(filters))
+                    newItems = sequenceParser.browse(subFolder, detectionMethod, filters)
```

Comment:
`sequenceParser.browse` browse the input folder but not recursively. And I don't see any code around to make the recursion.